packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -323,7 +386,13 @@ class ScaffoldState extends State<Scaffold> {
     if (_snackBars.isNotEmpty)
       _addIfNonNull(children, _snackBars.first._widget, _Child.snackBar);
 
-    _addIfNonNull(children, config.floatingActionButton, _Child.floatingActionButton);
+    if (config.floatingActionButton != null) {
+      Widget fab = new _FloatingActionButtonTransition(
+        key: new ValueKey<Key>(config.floatingActionButton.key),
```

Comment:
Without this key, when the transition from the old FAB to the new one starts, the two FABs get mixed up. The old one seems to end up with the new one's performance, until the new one grows in.